src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -131,5 +161,43 @@ private static bool IsValidValue(object value)
 
             return false;
         }
+
+        private static Resource GetOTelEnvVarResource()
+        {
+            var resource = Resource.Empty;
+
+            string envResourceAttributeValue = Environment.GetEnvironmentVariable(OTelResourceEnvVarKey);
+            if (envResourceAttributeValue != null)
+            {
+                var attributes = ParseResourceAttributes(envResourceAttributeValue);
+                return new Resource(attributes);
+            }
+
+            return resource;
+        }
+
+        private static IEnumerable<KeyValuePair<string, object>> ParseResourceAttributes(string resourceAttributes)
+        {
+            var attributes = new List<KeyValuePair<string, object>>();
+
+            if (string.IsNullOrEmpty(resourceAttributes))
```

Comment:
nit: There is a null check in `GetOTelEnvVarResource` if you made that a `string.IsNullOrEmpty` check instead then this wouldn't be needed here.